### PR TITLE
Simulate vision

### PR DESCRIPTION
**Pull request by vichik123**
This pull request is part of the vision branch. This branch simulates the drivetrain (as a tank drive system; later on as a swerve drive), and tells the user when the hub is in range.
It includes:
Subsystems: PhotonVisionModule, SimulateDrivetrain.
Commands: SimulateDrivetrainDefaultCommand.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+        public static final double CAMERA_HEIGHT = 0.767; // [m]
+        public static final double TARGET_HEIGHT_FROM_GROUND = 2.64; // [m]
+        public static final double CAMERA_PITCH = 30; // Pitch of the vision. [deg]
+        public static final double DIAG_FOV = 75; // Diagonal FOV. [deg]
+        public static final double LED_RANGE = 6; // Visible range of LEDs. [m]
+        public static final int CAM_RESOLUTION_WIDTH = 640; // Width of camera resolution. [pixel]
+        public static final int CAM_RESOLUTION_HEIGHT = 480; // Height of camera resolution. [pixel]
+        public static final double MIN_TARGET_AREA = 10; // Minimal area of target. [pixel^2]
+        public static final double TARGET_WIDTH = 1.36; // Width of vision target strip. [m]
+        public static final double TARGET_HEIGHT = 0.05; // Height of the vision target strip. [m]
```
Comment (adamv17) [inline src/main/java/frc/robot/Constants.java:28]: Why is this 0.05 shouldn't the target height be the TARGET_HEIGHT_FROM_GROUND?

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+
+        public static final Pose2d HUB_POSE = new Pose2d( // Position of the hub relative to the field.
+                new Translation2d(FIELD_LENGTH / 2, FIELD_WIDTH / 2), new Rotation2d());
+        public static final Transform2d CAMERA_TO_ROBOT = new Transform2d(
```
Comment (adamv17) [inline src/main/java/frc/robot/Constants.java:32]: How did you find this?

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
+import frc.robot.utils.commands.SimulateDrivetrainDefaultCommand;
 import frc.robot.valuetuner.ValueTuner;
 import webapp.Webserver;
 
+import java.util.Optional;
+
 public class RobotContainer {
   // The robot's subsystems and commands are defined here...
-    private final XboxController xbox = new XboxController(Ports.Controls.XBOX);
+    private final XboxController xbox = new XboxController(0);
```
Comment (barel13) [inline src/main/java/frc/robot/RobotContainer.java]: Why did you delete the constant?

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (adamv17) [reply src/main/java/frc/robot/Constants.java:28]: nvm I get it, it's the height of the strip, it is a bit confusing even with documentation

File: src/main/java/frc/robot/Constants.java (hunk 2 above)
Comment (vichik123) [reply src/main/java/frc/robot/Constants.java:32]: From CAD

**Review by adamv17 — APPROVED** [review]
Beautiful

**Review by barel13 — APPROVED** [review]
Good Good
